### DISCUSSION TITLE
Dex as a non-root user?

### DISCUSSION THREAD

**Discussion by davidmather** [dexidp/dex#2106] [Q&A]
Hi, 

I have an application that 100% requires non-root authentication over LDAP etc. and was wondering does Dex require root?
We currently use SSSD and it cannot run as a non-root user so are looking for alternatives.

/Dave

**Comment by sagikazarmark** [comment]
I'm not sure I understand the question to be honest. Are you asking whether the Dex process must run as root? Or are yo asking about privileges in LDAP?

**Reply by davidmather** [reply]
Both really, we have an application that runs the internet and uses LDAP/active directory. It has traditionally only ever run as root with SSSD for LDAP active directory stuff such as connecting to NFS fileshares within a customer's network. They would like to use kubernetes for scalability in times of high network load. SSSD cannot run as non-root at the moment and because of the nature of the application we have a requirement for running as non root. So we are looking at alternatives that might be able to run as non-root.
